From 7f6d34d154b9fe110c168a5b44083d8e35bb2068 Mon Sep 17 00:00:00 2001
From: "Bryn M. Reeves" <bmr@redhat.com>
Date: Mon, 10 Feb 2014 15:27:24 +0000
Subject: [PATCH 18/72] Add distupgrade plugin

Add a new plugin to collect data relating to completed or
attempted distribution upgrades. Currently supports the Red Hat
set of tools and files but other distributions can supply their
own filelists or override setup() to perform other distribution
specific actions.

Signed-off-by: Bryn M. Reeves <bmr@redhat.com>
---
 sos/plugins/distupgrade.py | 39 +++++++++++++++++++++++++++++++++++++++
 1 file changed, 39 insertions(+)
 create mode 100644 sos/plugins/distupgrade.py

diff --git a/sos/plugins/distupgrade.py b/sos/plugins/distupgrade.py
new file mode 100644
index 0000000..b45d099
--- /dev/null
+++ b/sos/plugins/distupgrade.py
@@ -0,0 +1,39 @@
+## Copyright (C) 2014 Red Hat, Inc., Bryn M. Reeves <bmr@redhat.com>
+
+### This program is free software; you can redistribute it and/or modify
+## it under the terms of the GNU General Public License as published by
+## the Free Software Foundation; either version 2 of the License, or
+## (at your option) any later version.
+
+## This program is distributed in the hope that it will be useful,
+## but WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+## GNU General Public License for more details.
+
+## You should have received a copy of the GNU General Public License
+## along with this program; if not, write to the Free Software
+## Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+import os
+from sos.plugins import Plugin, RedHatPlugin
+
+class DistUpgrade(Plugin):
+    """ Distribution upgrade data """
+
+    plugin_name = "distupgrade"
+
+    files = None
+
+    def setup(self):
+        self.add_copy_specs(list(self.files))
+
+class RedHatDistUpgrade(DistUpgrade, RedHatPlugin):
+
+    files = (
+        "/var/log/upgrade.log",
+        "/var/log/redhat_update_tool.log",
+        "/root/preupgrade/all-xccdf*",
+        "/root/preupgrade/kickstart"
+    )
+
+
-- 
1.9.3

